schemas/pci/pci-bus.yaml

```diff
@@ -22,139 +23,202 @@ description: |
+      Many PCI devices have only one function per device.
 
-  ranges:
-    oneOf:
-      - $ref: "/schemas/types.yaml#/definitions/flag"
-      - minItems: 1
-        maxItems: 32    # Should be enough
-        items:
-          minItems: 5
-          maxItems: 7
-          additionalItems: true
-          items:
-            - enum:
-                - 0x01000000
-                - 0x02000000
-                - 0x03000000
-                - 0x42000000
-                - 0x43000000
-                - 0x81000000
-                - 0x82000000
-                - 0x83000000
-                - 0xc2000000
-                - 0xc3000000
-
-  dma-ranges:
-    oneOf:
-      - type: boolean
-      - minItems: 1
-        maxItems: 32    # Should be enough
-        items:
-          minItems: 5
-          maxItems: 7
-          additionalItems: true
-          items:
-            - enum:
-                - 0x02000000
-                - 0x03000000
-                - 0x42000000
-                - 0x43000000
-
-  "#address-cells":
-    const: 3
-
-  "#size-cells":
-    const: 2
-
-  device_type:
-    const: pci
-
-  bus-range:
-    $ref: /schemas/types.yaml#/definitions/uint32-array
-    minItems: 2
-    maxItems: 2
-    items:
-      maximum: 255
-
-  "#interrupt-cells":
-    const: 1
-
-  interrupt-map: true
-#    minItems: 1
-#    maxItems: 88    # 22 IDSEL x 4 IRQs
-#    items:
-#      minItems: 6   # 3 addr cells, 1 PCI IRQ cell, 1 phandle, 1+ parent addr and IRQ cells
-#      maxItems: 16
-
-  interrupt-map-mask:
-    items:
-      - description: PCI high address cell
-        minimum: 0
-        maximum: 0xf800
-      - description: PCI mid address cell
-        const: 0
-      - description: PCI low address cell
-        const: 0
-      - description: PCI IRQ cell
-        minimum: 0
-        maximum: 7
-
-  linux,pci-domain:
-    $ref: /schemas/types.yaml#/definitions/uint32
-
-  max-link-speed:
-    $ref: /schemas/types.yaml#/definitions/uint32
-    enum: [ 1, 2, 3, 4 ]
-
-  num-lanes:
-    description: The number of PCIe lanes
-    $ref: /schemas/types.yaml#/definitions/uint32
-    enum: [ 1, 2, 4, 8, 16, 32 ]
-
-  reset-gpios:
-    description: GPIO controlled connection to PERST# signal
-    maxItems: 1
-
-  supports-clkreq:
-    type: boolean
-
-  aspm-no-l0s:
-    description: Disables ASPM L0s capability
-    type: boolean
-
-  vendor-id:
-    description: The PCI vendor ID
-    $ref: /schemas/types.yaml#/definitions/uint32
-
-  device-id:
-    description: The PCI device ID
-    $ref: /schemas/types.yaml#/definitions/uint32
-
-patternProperties:
-  "^[a-zA-Z][a-zA-Z0-9,+\\-._]{0,63}@1?[0-9a-f](,[0-7])?$":
-    type: object
     properties:
+      $nodename:
+        pattern: "^[a-zA-Z][a-zA-Z0-9,+\\-._]{0,63}@1?[0-9a-f](,[0-7])?$"
+
       compatible:
         contains:
           pattern: "^(pci[0-9a-f]{3,4},[0-9a-f]{1,4}|pciclass,[0-9a-f]{4,6})$"
+
+      device_type:
+        const: pci
```

Comment:
device_type only belongs in PCI bridge nodes.

Reply:
Ok... I thought that it applies for all PCI compatible devices.